### PR TITLE
Disable release creation workflow on forks

### PR DESCRIPTION
**Pull request by gbrunin**
I kept having this issue with our fork:
https://github.com/Matgenix/matminer/actions/runs/10141810059/job/28039988779

which I fixed by setting GITHUB_TOKEN and PAT explicitly. I'm not super familiar with the test.yml file so maybe there's a better fix (or maybe it should not be fixed), let me know.

### CODE REVIEW

**Comment by gbrunin** [comment]
Not sure why the 2 tests on datasets are failing with python 3.9

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -92,7 +92,8 @@ jobs:
       - test
     runs-on: ubuntu-latest
```
Comment (ml-evs) [inline .github/workflows/test.yml:93]: I think it's probably easier to just disable this on forks with something like (might need to move it into the steps)

```suggestion
    runs-on: ubuntu-latest
    if: github.repository == 'hackingmaterials/matminer'
```

This part of the release doesn't seem to be used much; in the past I've just manually made the release and its all seemed to work.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -91,9 +91,9 @@ jobs:
     needs:
       - test
     runs-on: ubuntu-latest
+    if: github.repository == 'hackingmaterials/matminer'
     env:
-      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      PAT: ${{ secrets.PAT }}
+      GITHUB_TOKEN: ${{ secrets.PAT }}
```
Comment (ml-evs) [inline .github/workflows/test.yml:96]:
```suggestion
      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Review by ml-evs — COMMENTED** [review]
Not sure why the test pypi build is failing, or why it is even triggering...

File: .github/workflows/test.yml (hunk 2 above)
Comment (ml-evs) [reply .github/workflows/test.yml:96]: Whoops, misread the original

**Review by ml-evs — APPROVED** [review]
I think the other failures are unrelated, so happy to merge this now if it fixes things at your end @gbrunin.

**Pull request merged**